app/notification/model/notification.py

```
@@ -40,53 +40,58 @@ def email_recipient(json_data: dict):
+                    return Notifier.send_application_reminder(notification)
+
+                case NotifyConstants.TEMPLATE_TYPE_EOI_PASS:
+                    current_app.logger.info(f"Validating template type: {notification.template_type})")
+                    return Notifier.send_submitted_eoi(
+                        notification=notification,
+                        template_name=NotifyConstants.TEMPLATE_TYPE_EOI_PASS,
+                    )
+
+                case NotifyConstants.TEMPLATE_TYPE_EOI_PASS_W_CAVEATS:
+                    current_app.logger.info(f"Validating template type: {notification.template_type})")
+                    return Notifier.send_submitted_eoi(
+                        notification=notification,
+                        template_name=NotifyConstants.TEMPLATE_TYPE_EOI_PASS_W_CAVEATS,
+                    )
+
+                case NotifyConstants.TEMPLATE_TYPE_ASSESSMENT_APPLICATION_ASSIGNED:
+                    current_app.logger.info(f"Validating template type: {notification.template_type})")
+                    return Notifier.send_assessment_assigned(
+                        notification=notification,
+                    )
+
+                case NotifyConstants.TEMPLATE_TYPE_ASSESSMENT_APPLICATION_UNASSIGNED:
+                    current_app.logger.info(f"Validating template type: {notification.template_type})")
+                    return Notifier.send_assessment_unassigned(
+                        notification=notification,
+                    )
+
+                case "NOTIFICATION" | "AWARD":
+                    return f"Currently {notification.template_type} service is not available."  # noqa
+
+                case _:
+                    current_app.logger.exception(f"Incorrect template type {notification.template_type}")
+                    return template_type_error(json_data)
+
+        except Exception as e:
```

Minor point, but this will capture all exceptions, and later you raise a KeyError (even if that wasn't the original exception). Maybe something like this : 

```
except Exception as e:
    current_app.logger.error("An exception occurred while selecting email template to send", e)
    raise Exception("An error occurred while selecting email template to send") from e
```

---

nice let me do the fix yeah I also thought that I am hiding the exception